js-api/src/chem.ts

```diff
@@ -193,23 +193,25 @@ export namespace chem {
       return Sketcher.isEmptyMolfile(molFile);
     }
 
-    /** Sets the molecule, supports either SMILES or MOLBLOCK formats */
-    async setMolecule(molString: string, substructure: boolean = false) {
+    /** Sets the molecule, supports either SMILES, SNARTS or MOLBLOCK formats */
+    setMolecule(molString: string, substructure: boolean = false) {
       if (substructure)
         this.setSmarts(molString);
       else if (isMolBlock(molString))
         this.setMolFile(molString);
       else {
         this._smiles = molString;
         //setting molFile instead of smiles to draw in coordinates similar to dataframe cell
-        const mol = (await grok.functions.call('Chem:getRdKitModule')).get_mol(molString, '{"mergeQueryHs":true}');
-        if (!mol.has_coords())
-          mol.set_new_coords();
-        mol.normalize_depiction(1);
-        mol.straighten_depiction(false);
-        this._molfile = mol.get_molblock();
-        this.setMolFile(this._molfile);
-        mol?.delete();
+        grok.functions.call('Chem:getRdKitModule').then((rdkit: any) => {
```

Comment:
Consider moving this code to Chem and simply calling a function instead of working with RDMol directly